fabtests/benchmarks/benchmark_shared.c

```diff
@@ -381,6 +381,78 @@ int pingpong_rma(enum ft_rma_opcodes rma_op, struct fi_rma_iov *remote)
 	return 0;
 }
 
+int rma_tx_completion(enum ft_rma_opcodes rma_op, struct fi_rma_iov *remote)
+{
+	int ret, i;
+	size_t inject_size = fi->tx_attr->inject_size;
+
+	ret = fi_getopt(&ep->fid, FI_OPT_ENDPOINT, FI_OPT_INJECT_RMA_SIZE,
+			&inject_size, &(size_t){sizeof inject_size});
+	if (ret && ret != -FI_ENOPROTOOPT) {
+		FT_PRINTERR("fi_getopt(FI_OPT_INJECT_RMA_SIZE)", ret);
+		return ret;
+	}
+
+	if (inject_size_set)
+		inject_size = opts.inject_size;
+
+	if (ft_check_opts(FT_OPT_ENABLE_HMEM))
+		inject_size = 0;
+
+	if (ft_check_opts(FT_OPT_VERIFY_DATA)) {
+		FT_ERR("Verify Data Is Not Supported");
+		return EXIT_FAILURE;
+	}
+
+	ret = ft_sync();
+	if (ret)
+		return ret;
+
+	if (opts.dst_addr) {
+		for (i = 0; i < opts.iterations + opts.warmup_iterations; i++) {
+			if (i == opts.warmup_iterations)
+				ft_start();
+
+			if (opts.transfer_size <= inject_size)
+				ret = ft_inject_rma(rma_op, remote, ep,
+						    remote_fi_addr,
+						    opts.transfer_size);
+			else
+				ret = ft_tx_rma(rma_op, remote, ep, remote_fi_addr,
+						opts.transfer_size, &tx_ctx);
+			if (ret)
+				return ret;
+		}
+
+		ft_stop();
+		if (opts.machr)
+			show_perf_mr(opts.transfer_size, opts.iterations, &start, &end, 1,
+				opts.argc, opts.argv);
+		else
+			show_perf(NULL, opts.transfer_size, opts.iterations, &start, &end, 1);
+
+		/* Inform RMA target that the test has ended */
+		ret = ft_sync();
+		if (ret)
+			return ret;
+	} else {
+		if (rma_op == FT_RMA_WRITEDATA) {
```

Comment:
I think you can remove this check

`ft_rx_rma` will automatically poll the last byte for `fi_write`
https://github.com/ofiwg/libfabric/blob/4cadc72f49e0648b6cab6d050edcd3ed36570e0c/fabtests/common/shared.c#L2617-L2619

Reply:
I rely on the ft_sync at the end to let the receiver know the sender is complete.  I didn't want that check.

```
		/* Wait for writer to complete */
		ret = ft_sync();
```

Reply:
The only reason I call `ft_rx_rma()` I kept trying to short-circuit that method, but I kept running into issues handling completions... so I figured it would be easiest to just stick it there.

Reply:
> I kept running into issues handling completions

Likely because the next expected completion count is a global variable

Would you mind adding a comment along these lines?
> Drain fi_writedata CQEs on the receiver side